runtime/autoload/man.vim

```diff
@@ -97,6 +97,23 @@ function! s:read_page(sect, name) abort
     silent keepjumps 1delete _
   endwhile
   setlocal filetype=man
+  call man#create_toc()
+endfunction
+
+function! man#create_toc() abort
+  if !exists('b:man_toc')
+    " Cache the TOC since user might switch in and out between the man pages.
+    let b:man_toc = []
+    for lnum in range(1, line('$'))
+      let text = getline(lnum)
+      if text =~# '^\%(\%(\S.*\)\=\S\|\%1l.*\| \{3\}\S.*\)$'
+        let text = substitute(getline(lnum), '\s\+', ' ', 'g')
```

Comment:
There were cases of right-aligned text or "table" column headers on some lines.  It would make the horizontal scrolling long.